hail/python/hail/genetics/pedigree.py

```diff
@@ -201,8 +200,40 @@ def read(cls, fam_path, delimiter='\\s+') -> 'Pedigree':
         :rtype: :class:`.Pedigree`
         """
 
-        jrep = Env.hail().methods.Pedigree.read(fam_path, Env.hc()._jhc.sFS(), delimiter)
-        return Pedigree._from_java(jrep)
+        trios = []
+        missing_sex_count = 0
+        missing_sex_values = set()
+        with Env.fs().open(fam_path) as file:
+            for line in file:
+                split_line = re.split(delimiter, line.strip())
+                num_fields = len(split_line)
+                if num_fields != 6:
+                    raise FatalError("Require 6 fields per line in .fam, but this line has {}: {}".format(num_fields, line))
+                (fam, kid, dad, mom, sex, _) = tuple(split_line)
+                # 1 is male, 2 is female, 0 is unknown.
+                is_female = sex == "2" if sex == "1" or sex == "2" else None
+
+                if is_female is None:
+                    missing_sex_count += 1
+                    missing_sex_values.add(sex)
+
+                trio = Trio(kid,
+                            fam if fam != "0" else None, 
```

Comment:
Can you confirm that the docs state the missing value is "0"? We should in the future parameterize this at some point, but not critical.